### PR TITLE
Integrate support for no-access Idps with all functionality

### PR DESCRIPTION
**Pull request by Badlapje**
Ensured idps without access are visible & styled appropriately

### CODE REVIEW

**Review by MKodde — COMMENTED** [review]
This time I started with a visual/functional inspection

zep-vs-fox

The PR title suggests all no-access functionalities are covered in this PR. However, the request access screen is not yet styled. Although it seems to function very well!

File: theme/base/javascripts/wayf/noAccess/handleCancelNoAccess.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import {toggleVisibility} from '../../utility/toggleVisibility';
+
+export const handleCancelNoAccess = (parentSection, noAccessSection) => {
+  toggleVisibility(noAccessSection);
+  toggleVisibility(parentSection);
+  // todo show banner
```
Comment (MKodde) [inline theme/base/javascripts/wayf/noAccess/handleCancelNoAccess.js]: What banner should be shown here?

File: theme/base/javascripts/wayf/handleClickingDisabledIdp.js [hunk 2]
```diff
@@ -14,6 +19,15 @@ export const handleClickingDisabledIdp = (element) => {
   li.innerHTML = '';
   li.insertAdjacentElement('afterbegin', cloneOfIdp);
 
+  // if successModal is visible, hide it
+  hideSuccessMessage();
+
+  // add idp entityId
+  setHiddenFieldValues(element, form);
```
Comment (MKodde) [inline theme/base/javascripts/wayf/handleClickingDisabledIdp.js:26]: Perhaps a rudimentary validation step in between these? I was able to submit the form empty, and with invalid values (non-email in email field for example)

**Review by MKodde — COMMENTED** [review]
Not much to complain about the code changes! Things are looking good!

File: theme/base/javascripts/wayf/noAccess/handleCancelNoAccess.js (hunk 1 above)
Comment (Badlapje) [reply theme/base/javascripts/wayf/noAccess/handleCancelNoAccess.js]: this method was removed, along with the comment, seems to be a mixup with github showing an outdated file?

**Comment by Badlapje** [comment]
💯 i agree with those full-width input fields.  I stuck to the design for now, but i'm really hoping that when it's tested it'll be a common conclusion!

File: theme/base/javascripts/wayf/handleClickingDisabledIdp.js (hunk 2 above)
Comment (Badlapje) [reply theme/base/javascripts/wayf/handleClickingDisabledIdp.js:26]: added html5 validation, as we have validation in the backend and the requirements are very simple.  Motivation is not required atm (but waiting on feedback from Edwin/Arnout there)

**Review by MKodde — APPROVED** [review]
Great step forward!

1. I see the `displayUnconnectedIdpsWayf` once again functions, at least it triggers the display of the form. Not fully sure that is the expected behaviour, but that can be verified with PM
2. Thanks for adding the HTML validation, that should prove sufficient. I guess therre is little/no way to manipulate the way the validation tooltips are rendered in browser? In Firefox the tooltip is rendered in a somewhat wonky manner.
3. The unconnected styling is still not that different from the connected IdP styling. 
4. Completely unrelated to your changes: when keeb-navigating through the list, it would be nice to have the currently active/selected IdP have the hover state that is also used when hovering over with mouse.

Feel free to merge and address these points sometime later.

**Comment by Badlapje** [comment]
1. I've made a separate commit for the displayUnnconnectedIdpsWayf !936
2. I've added error messages & styling as per the styleboard now 😄 
3. I know, but it is as per the styleboard / designs.  I did forget the box-shadow which i've added now.
4. I agree, but followed the style board.  I'll ask Edwin/Arnout about our three known styling disagreements ;-)

**Pull request merged**